website/js/src/statistics.js

```diff
@@ -0,0 +1,99 @@
+ */
+const navigate = (url, paramKey, paramValue) =>
+    window.location.href = `${url}?${paramKey}=${paramValue}`;
+
+/**
+ * Populate the given element with the given items using the given template.
+ * It also assigns a click event handler to each item.
+ *
+ * @param {Object[]} items - An array of objects,
+ * where each object represents a row of data with properties corresponding to different data attributes.
+ * @param {string} elementId - The ID of the HTML element that will be populated with items.
+ * @param {function(Object): string} template - A function that takes an item (object)
+ * as an argument and returns an HTML string that represents the item.
+ * @param {function(string): void} onClick - A function that will be called when an item
+ * is clicked.
+ * It receives the ID of the clicked item as an argument.
+ */
+const populateElement = (items, elementId, template, onClick) => {
+    const element = document.getElementById(elementId);
+    element.innerHTML = '';
+    items.forEach(item => element.innerHTML += template(item));
+
+    // Add click listener to each row
+    element.addEventListener('click', e => {
+        const itemRow = e.target.closest('tr');
+        if (itemRow) {
+            const itemId = itemRow.dataset.id;
+            onClick(itemId);
+        }
+    });
+}
+
+/**
+ * Generates the best and worst items based on the provided criteria and populates the specified elements with them.
+ *
+ * @param {Object[]} data - An array of objects,
+ * where each object represents a row of data with properties corresponding to different data attributes.
+ * @param {string} sortProperty - The property name to sort the data by.
+ * It should be a key that exists in each object within the data array.
+ * @param {function(Object[], string): void} populateFn - A function that takes an array of objects and an element ID as arguments,
+ * and populates the specified HTML element with the provided data.
+ * @param {string} bestElementId -
+ * The ID of the HTML element to be populated with the best items based on the sorting criteria.
+ * @param {string} worstElementId -
+ * The ID of the HTML element to be populated with the worst items based on the sorting criteria.
+ */
+function generateBestWorst(data, sortProperty, populateFn, bestElementId, worstElementId) {
+    const sortedData = data.sort((a, b) => a[sortProperty] - b[sortProperty]);
+    const bestItems = sortedData.slice(0, 5);
+    const worstItems = sortedData.slice(-5).reverse();
+
+    populateFn(bestItems, bestElementId);
+    populateFn(worstItems, worstElementId);
+}
+
+document.addEventListener('DOMContentLoaded', async () => {
+    // Load the data from the csv files
+    const trainData = await d3.csv(TRAIN_DATA_CSV_PATH);
+    const stopData = await d3.csv(STOP_DATA_CSV_PATH);
+
+    // Generate the best and worst trains
+    generateBestWorst(
+        trainData,
+        'median_arrival_delay',
```

Comment:
worst trains are computed with the median and worst stops with the average?

Reply:
Well I'm not really the one to ask for this hahaha, that's the way it was in the original file

Reply:
Ok, can we make both with the median delay?

Reply:
Ok, but is it in the dataset for the stops?

Reply:
I checked and it is so I'm going to change